### ISSUE TITLE
Marionette Application needs a "root"

### ISSUE THREAD

**Issue by samccone** [marionettejs/backbone.marionette#2279]
With the deprecation of the application regionManager on the Application object, we need a way for people to "kick off" their view stack, scoped to an application.

```js

app = new Mn.Application({rootEl: "#my-app"})

app.onStart () =>
  app.root.show(new MyViewClass)
```

This is something that we can add to minor and allow people to start migrating away from a region manager today, it will also provide a better answer to people when they ask ... what do i do now since region managers were deprecated.

**Comment by jmeas** [comment]
:+1:

Two things:

1. It should be called `rootElement` or `rootEl`.
2. It should be `body` by default.

**Comment by jfairbank** [comment]
`rootEl`

**Comment by jmeas** [comment]
Also, I like this best if the `rootElement` becomes something more like Ember's outlet, vs. a region.

In Marionette, there's no easy way to get access to `app.root` from within a route. One option is accessing the app directly. Another option is Radio. A third option is the way that Ember does it. Consider this URL (from within a Route)

```js
show() {
  this.showIn('root', new View());
}
```

`'root'` would use the [resolver](https://github.com/marionettejs/backbone.marionette/issues/2207) pattern to find the region in the DOM tree hierarchy. The Router would keep track of these nested regions.

**Comment by samccone** [comment]
:+1: Priority bump just based on the number of questions that we have been getting regarding this

**Comment by jasonLaster** [comment]
:dancers:

**Comment by ahumphreys87** [comment]
fixed via #2763
